cmd/autoscaler/main.go

```diff
@@ -63,89 +63,53 @@ var (
 )
 
 func main() {
-	flag.Parse()
+	// Set up the components.
 
-	loggingConfigMap, err := configmap.Load("/etc/config-logging")
-	if err != nil {
-		log.Fatalf("Error loading logging configuration: %v", err)
-	}
-	loggingConfig, err := logging.NewConfigFromMap(loggingConfigMap)
-	if err != nil {
-		log.Fatalf("Error parsing logging configuration: %v", err)
-	}
+	flag.Parse()
 
-	var atomicLevel zap.AtomicLevel
-	logger, atomicLevel := logging.NewLoggerFromConfig(loggingConfig, component)
+	logger, atomicLevel := setupLogger()
 	defer logger.Sync()
 
-	// set up signals so we handle the first shutdown signal gracefully
-	stopCh := signals.SetupSignalHandler()
-	statsCh := make(chan *autoscaler.StatMessage, statsBufferLen)
+	// Set up signals so we handle the first shutdown signal gracefully
+	stopCh, statsCh := setupChannels()
 
-	cfg, err := clientcmd.BuildConfigFromFlags(*masterURL, *kubeconfig)
-	if err != nil {
-		logger.Fatalw("Error building kubeconfig", zap.Error(err))
-	}
+	kubeClientSet, servingClientSet, scaleClient := setupClients(logger, stopCh)
 
-	kubeClientSet, err := kubernetes.NewForConfig(cfg)
-	if err != nil {
-		logger.Fatalw("Error building kubernetes clientset", zap.Error(err))
-	}
+	dynConfig := setupAutoscalerConfig(logger)
```

Comment:
naming this autoscalerConfig or scalerConfig makes more sense.

Reply:
Done.